src/AWS.Deploy.Orchestration/Orchestrator.cs

```diff
@@ -73,6 +91,15 @@ public async Task<List<Recommendation>> GenerateDeploymentRecommendations()
 
         public async Task DeployRecommendation(CloudApplication cloudApplication, Recommendation recommendation)
         {
+            if (_interactiveService == null)
+                throw new InvalidOrchestratorException($"{nameof(_interactiveService)} is null as part of the orchestartor object");
+            if (_cdkManager == null)
+                throw new InvalidOrchestratorException($"{nameof(_cdkManager)} is null as part of the orchestartor object");
+            if (_cdkProjectHandler == null)
+                throw new InvalidOrchestratorException($"{nameof(_cdkProjectHandler)} is null as part of the orchestartor object");
+            if (_session == null)
+                throw new InvalidOrchestratorException($"{nameof(_session)} is null as part of the orchestartor object");
```

Comment:
Throw `InvalidOperationException`, we don't need custom exception. `InvalidOperationException` is purposefully created for invalid state.

https://docs.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-5.0

Reply:
Fixed